lib/puppet_x/sqlserver/sql_connection.rb

```diff
@@ -90,25 +90,26 @@ def win32_exception
     end
 
     class ResultOutput
+
       attr_reader :exitstatus, :error_message, :raw_error_message
 
-      def initialize(has_errors, error_message)
+      def initialize(has_errors, error_message, connection)
         @exitstatus = has_errors ? 1 : 0
-        if error_message
-          @raw_error_message = error_message
-          @error_message = parse_for_error(error_message)
+
+        unless connection.nil?
+          unless connection.Errors.count == 0
+            @error_message = connection.Errors(0).Description
+          end
+        end
```

Comment:
Could this be simpler as:

```ruby
@error_message = connection.Errors(0).Description if connection && !connection.Errors.empty?
```

Reply:
The Errors collection is never a null property, it just throws an error if you try to call an index if it's not storing any errors, which is why I'm calling the ```count``` property first. I could probably get away with ```@error_message = connection.Errors(0).Description if connection && !connection.Errors.count == 0```

Reply:
The `.empty?` I suggested should generally be the same as `.count == 0` ... unless this is a COM enumerable?  In which case using `.count == 0` is probably the correct thing to do.

Reply:
Worse than I thought. This object just doesn't respond to ```.empty?```. It's a OLE object so I'm guess that's why.

Reply:
It's probably not an Enumerable object.

Reply:
That's what I was saying @glennsarti ;0 

But not only is it not a COM enumerable like [IEnumVariant](https://msdn.microsoft.com/en-us/library/windows/desktop/ms221053\(v=vs.85\).aspx) ... which Ruby cannot enumerate ... but the interface doesn't have enumeration methods anyhow.

https://docs.microsoft.com/en-us/sql/ado/reference/ado-api/errors-collection-properties-methods-and-events